ui/src/message_popup/author/questions_page.jsx

```diff
@@ -0,0 +1,162 @@
+import Divider from 'material-ui/Divider';
+import IconButton from 'material-ui/IconButton';
+import Paper from 'material-ui/Paper';
+import MenuItem from 'material-ui/MenuItem';
+import {Card, CardHeader, CardText} from 'material-ui/Card';
+
+import SearchIcon from 'material-ui/svg-icons/action/search';
+import AddIcon from 'material-ui/svg-icons/content/add';
+import ArchiveIcon from 'material-ui/svg-icons/content/archive';
+
+import NavigationAppBar from '../../components/navigation_app_bar.jsx';
+import QuestionButton from './question_button.jsx';
+
+import {allQuestions} from '../questions.js';
+import {withStudents, withLearningObjectiveAndIndicator} from '../transformations.jsx';
+
+export default React.createClass({
+  displayName: 'QuestionsPage',
+
+  getInitialState(){
+    return ({
+      questions: allQuestions,
+      showArchivedQuestions: false
+    });
+  },
+  
+  onNewQuestion(){
+    Routes.navigate(Routes.messagePopupAuthorQuestionsNewPath());
+  },
+
+  onArchivedQuestions(){
+    Routes.navigate(Routes.messagePopupAuthorArchivedQuestionsPath());
+  },
+
+  onSearchBarChange(event){
+    const value = event.target.value.toLowerCase().trim();
+    if(value === ''){
+      this.setState({questions: allQuestions});
+    }else{
+      var questions = _.clone(allQuestions);
+      questions = questions.filter(questionOriginal => {
+        const question = withStudents([withLearningObjectiveAndIndicator(questionOriginal)])[0];
+        var questionString = question.id + " " + question.text + " " + question.learningObjective.key + " " + question.learningObjective.text + " " + question.learningObjective.competencyGroup + " " + question.indicator.text;
+        if(_.has(question, 'students')) _.forEach(question.students, student => questionString += " " + student.id + " " + student.name);
+        _.forEach(question.examples, example => questionString += " " + example);
+        _.forEach(question.nonExamples, example => questionString += " " + example);
+        return questionString.toLowerCase().includes(value);
+      });
+      questions = questions.sort(questionOriginal => {
+        const question = withStudents([withLearningObjectiveAndIndicator(questionOriginal)])[0];
+        var questionString = question.id + " " + question.text + " " + question.learningObjective.key + " " + question.learningObjective.text + " " + question.learningObjective.competencyGroup + " " + question.indicator.text;
+        if(_.has(question, 'students')) _.forEach(question.students, student => questionString += " " + student.id + " " + student.name);
+        _.forEach(question.examples, example => questionString += " " + example);
+        _.forEach(question.nonExamples, example => questionString += " " + example);
+        return questionString.split(value).length;
+      });
+      this.setState({questions});
+    }
+  },
+
+  render(){
+    var archivedQuestions = [];
+    archivedQuestions.push({
```

Comment:
Even for testing locally, please split test data outside of the `render` method into another file or at least a constant at the top of the component.  Having data like this inside `render` can only lead to accidentally merging it :)